src/quantilealgs.jl

```diff
@@ -30,93 +30,97 @@ quantile_bisect(d::ContinuousUnivariateDistribution, p::Float64) =
 #   Distribution, with Application to the Inverse Gaussian Distribution
 #   http://www.statsci.org/smyth/pubs/qinvgaussPreprint.pdf
 
-function quantile_newton(d::ContinuousUnivariateDistribution, p::Float64, xs::Float64=mode(d), tol::Float64=1e-12)
-    if 0.0 < p < 1.0
+function quantile_newton(d::ContinuousUnivariateDistribution, p::Real, xs::Real=mode(d), tol::Real=1e-12)
+    T = promote_type(partype(d), typeof(p), typeof(xs), typeof(tol))
```

Comment:
I don't think the tolerance needs to be part of the type determination. It can be convenient to specify the tolerance as a `Float64` without promoting all the other parameters.